### ISSUE TITLE
dynamic icon name after V4.0.0-alpha

### ISSUE THREAD

**Issue by Romanior** [martndemus/ember-font-awesome#173]
Hello, is any way to use dynamic icon name like `{{fa-icon myVar}}` ? It seems, that can not find any way after V4.0.0-alpha and its changes. 

From my understanding this is a common use-case, e.g. render the list with icons.
Thanks.

**Comment by cibernox** [comment]
That case is explicitly allowed, there is tests that assert that it works and is 100% a valid usage, you should have no problems with that.

Can you provide a reproduction?

**Comment by Romanior** [comment]
indeed, no matter how hard I tried to isolate issue, it is working on blank project with same dependencies. Let's keep this open for a while if someone else would face it, then I will close it.
Thanks.

**Comment by paulcwatts** [comment]
I created a reproduction for this: https://github.com/paulcwatts/ember-font-awesome-dyn-icon

Note the bug only happens with environment=production:

1. Clone the repository: `git clone https://github.com/paulcwatts/ember-font-awesome-dyn-icon.git`
2. `cd ember-font-awesome-dyn-icon`
3. `ember serve`
4. View `http://localhost:4200`, the icon should appear.
5. `ember serve --production=environment`
6. View `http://localhost:4200`, the icon doesn't appear.

I wanted to write a unit test for it, but it doesn't appear that unit tests work with environment=production?

**Comment by jrjohnson** [comment]
Thanks @paulcwatts. `production` is the only environment where icons are pruned when not used and it is proving to be difficult to write tests for this and this is why tests are likely passing even with this break.

**Comment by paulcwatts** [comment]
I updated my repro. With a bit more testing, this appears to have regressed between alpha.4 and alpha.5.

**Comment by cibernox** [comment]
@paulcwatts can you check alpha 10?

**Comment by paulcwatts** [comment]
@cibernox The above repo currently uses alpha 10, and successfully reproduces the issue.

**Comment by Romanior** [comment]
@paulcwatts Good catch, thanks!

**Comment by buschtoens** [comment]
Getting the same error over in Ember Light Table, but only with Ember versions below 2.12.

https://travis-ci.org/offirgolan/ember-light-table/jobs/318468250#L578
